src/main.rs

```diff
@@ -47,9 +49,26 @@ const VC_EXPIRATION_PERIOD_NS: u64 = 15 * MINUTE_NS;
 // End of year 2024 as UNIX timestamp.
 const EOY_2024_TIMESTAMP_S: u32 = 1735685999;
 
+// Internal container of per-event data.
+#[derive(CandidType, Clone, Deserialize)]
+struct EventRecord {
+    pub created_timestamp_s: u32,
+    pub code: String,
```

Comment:
maybe describe in a comment what is the meaning/function of `code`-field?

Reply:
Good idea, done!

Reply:
Much better, thanks!